test/fixtures/tags-config.yml

```diff
@@ -0,0 +1,3 @@
+tag_protection:
+  - pattern: "v1"
```

Comment:
i don think including `pattern` here is helpful since it is the only property. lets simplify to just be the pattern